libraries/JGUZDV.Blazor.Components/src/Inputs/MultiSelect.razor

```diff
@@ -0,0 +1,132 @@
+            {
+                <span class="text-muted small text-truncate align-self-center" title="@ItemTemplate(Value)">
+                    <span> (@ItemTemplate(Value))</span>
+                </span>
+            }
+        </div>
+        <div class="ms-1">
+            @if (_isOpen)
+            {
+                <i class="fa fa-fw fa-minus align-self-center"></i>
+            }
+            else
+            {
+                <i class="fa fa-fw fa-plus align-self-center"></i>
+            }
+        </div>
+    </div>
+
+    @if (_isOpen)
+    {
+        <div class="fade-in">
+            <MultiSelectBody Items="@Items"
+                             Values="@Values"
+                             ValuesChanged="@ValuesChanged"
+                             Value="@Value"
+                             ValueChanged="@ValueChanged"
+                             FilterLabel="@FilterLabel"
+                             Filter="@Filter"
+                             ItemsContainerClass="@ItemsContainerClass"
+                             ItemKey="@ItemKey"
+                             ItemTemplate="@ItemTemplate"
+                             AllOptionLabel="@AllOptionLabel"
+                             UseSingleSelectionMode="@UseSingleSelectionMode"
+                             T="T"
+                             TKey="TKey">
+            </MultiSelectBody>
+        </div>
+    }
+</div>
+@code {
+    [Parameter]
+    [EditorRequired]
+    public List<T> Items { get; set; } = default!;
+
+    [Parameter]
+    public RenderFragment<T>? ItemTemplate { get; set; } = default!;
+
+    [Parameter]
+    [EditorRequired]
+    public Func<T, string> ItemName { get; set; } = default!;
+
+    [Parameter]
+    [EditorRequired]
+    public Func<T, TKey> ItemKey { get; set; } = default!;
+
+    [Parameter]
+    public List<T>? Values { get; set; } = default!;
+
+    [Parameter]
+    public EventCallback<List<T>> ValuesChanged { get; set; }
+
+    [Parameter]
+    public T? Value { get; set; } = default!;
+
+    [Parameter]
+    public EventCallback<T> ValueChanged { get; set; }
+
+    [Parameter]
+    [EditorRequired]
+    public string Title { get; set; } = default!;
+
+    [Parameter]
+    public string? AllOptionLabel { get; set; }
+
+    [Parameter]
+    public string? HostClass { get; set; }
+
+    [Parameter]
+    public string? FilterLabel { get; set; }
+
+    [Parameter]
+    public Func<(T item, string query), bool>? Filter { get; set; }
+
+    [Parameter]
+    public string? ItemsContainerClass { get; set; } = "row row-cols-1 mt-2";
+
+    [Parameter]
+    public bool UseSingleSelectionMode { get; set; }
+
+    private bool _isOpen;
+
+    protected override void OnParametersSet()
+    {
+        base.OnParametersSet();
+
+        if (FilterLabel != null && Filter == null)
+            Filter = x => ItemName(x.item).Contains(x.query, StringComparison.OrdinalIgnoreCase);
+
+        if (ItemTemplate == null)
```

Comment:
Ich frage mich grade, ob das in OnParameterSet oder OnInitialized sein sollte - ist ein RenderFragment "statisch"?

Reply:
Der Consumer kanns ja ändern und die Komponente könnte recycled werden.